### DISCUSSION TITLE
Supported MySQL Versions

### DISCUSSION THREAD

**Discussion by dmitrymaslov99** [sidorares/node-mysql2#1519] [General]
Which versions the library supports? For example, the python library supports 4.1 + (https://pypi.org/project/PyMySQL/0.8.0/)

**Comment by sidorares** [comment]
There is almost no "major" versions of mysql protocol (with some exceptions), mostly a core + features that client and server can optionally support and indicate that via capabilities flags.

Need to double check, but from the back of my head they are (client side):

- text based query protocol
- prepared statements (aka binary protocol)
- multiple result sets
- large packet size
- compression and ssl
- authentication plugins
- authentication switch
- connection attributes
- client session state tracking
- automatic charset conversion
- built in `mysql_native_password` and `caching_sha2_password` auth plugins ( but no support for insecure pre-mysql 4.1 auth )

There is some partial support for binlog replication protocol but there hasn't been active development for it for quite some time. Happy to resume if there is an interest

There is server side of the protocol for some ( but not all ) of the features mentioned above

Might have missed something, feel free to ask for a particular feature and I'll try to confirm if supported or not
